File: 11-ContainerWithMostWater/11-ContainerWithMostWater.cpp
// Last updated: 3/26/2026, 1:28:12 PM
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n = height.size();
        int maxArea = 0;
        int l = 0;
        int r = n-1;
        while(l<r){
            int h = min(height[l],height[r]);
            maxArea = max(maxArea,h*(r-l));
            if(h == height[l]){
                l++;
            }else{
                r--;
            }
        }

        return maxArea;
    }
};

File: 11-ContainerWithMostWater/11-ContainerWithMostWater.py
# Last updated: 3/26/2026, 1:28:07 PM
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l = 0
        r = len(height)-1
        maxlen = 0
        while(l<r):
            he = min(height[l],height[r])
            maxlen = max(maxlen,he*(r-l))
            if(he == height[l]):
                l+=1
            else:
                r-=1
        return maxlen
        